dashboard/pages/explain_distsql.tsx

```diff
@@ -0,0 +1,239 @@
+  height: 80vh;
+`
+
+const position = {
+  x: 200,
+  y: 100,
+}
+
+const nodeTypes = { node: NodeType }
+
+function getColor() {
+  return (
+    "hsl(" +
+    360 * Math.random() +
+    "," +
+    (25 + 70 * Math.random()) +
+    "%," +
+    (85 + 10 * Math.random()) +
+    "%)"
+  )
+}
+
+function getStyle() {
+  return {
+    background: getColor(),
+    height: 50,
+    width: 200,
+    border: "0.5px solid black",
+    padding: "5px",
+    "border-radius": "5px",
+  }
+}
+
+function layoutElements(nodeList: any, edgeList: any) {
+  const g = new dagre.graphlib.Graph()
+  g.setGraph({
+    marginx: 20,
+    marginy: 20,
+  })
+  g.setDefaultEdgeLabel(function () {
+    return {}
+  })
+  for (var i = 0; i < nodeList.length; i++) {
+    g.setNode(nodeList[i]["id"], { width: 200, height: 100 })
+  }
+  for (var i = 0; i < edgeList.length; i++) {
+    g.setEdge(edgeList[i]["source"], edgeList[i]["target"])
+  }
+  dagre.layout(g)
+  for (var i = 0; i < nodeList.length; i++) {
+    const node = nodeList[i]
+    node["position"] = {
+      x: g.node(node["id"]).x - 500 / 2,
+      y: g.node(node["id"]).y - 50 / 2,
+    }
+  }
+}
+
+var visited = new Set()
+var stageToNode: { [key: string]: number } = {}
+var nodeStagePairs: number[][] = []
+
+function parseSubElements(root: any, style: any, nodeList: any, edgeList: any) {
+  if (root["children"].length == 0) return
+  for (var i = 0; i < root["children"].length; i++) {
+    const child = root["children"][i]
+    var edge = {
+      id: "e" + root["plan_node_id"] + "-" + child["plan_node_id"],
+      source: root["plan_node_id"].toString(),
+      target: child["plan_node_id"].toString(),
+      type: "smoothstep",
+    }
+    edgeList.push(edge)
+    if (visited.has(child["plan_node_id"])) continue
+    var node = {
+      id: child["plan_node_id"].toString(),
+      data: {
+        label: "#" + child["plan_node_id"] + " " + child["plan_node_type"],
+        content: Object.values(child["schema"]),
+      },
+      position: position,
+      type: "node",
+      style: style,
+    }
+    if (child["source_stage_id"] != null) {
+      nodeStagePairs.push([child["plan_node_id"], child["source_stage_id"]])
+    }
+    parseSubElements(child, style, nodeList, edgeList)
+    nodeList.push(node)
+  }
+}
+
+type PlanNode = {
+  plan_node_id: number
+  plan_node_type: string
+  schema: [any]
+  children: [PlanNode]
+  source_stage_id: number
+}
+
+type Stage = {
+  root: PlanNode
+  children: [number]
+  source_stage_id: number
+}
+
+function parseElements(input: any) {
+  var nodeList: Node[] = []
+  var edgeList: Edge[] = []
+  var stages: { [key: number]: Stage } = input["stages"]
+  for (const [key, value] of Object.entries(stages)) {
+    const root: PlanNode = value["root"]
+    stageToNode[key] = root["plan_node_id"]
+    var style = getStyle()
+    var node = {
+      id: root["plan_node_id"].toString(),
+      data: {
+        label: "#" + root["plan_node_id"] + " " + root["plan_node_type"],
```

Comment:
I'd prefer using format string like:

```
`#${root.planNodeId} ${root.planNodeType}`
```

Also it would be good to follow the naming convention (camelCase)

Reply:
I think these keys and naming conventions are derived from the JSON input